test/src/userobjects/InterfaceValueUO_QP.C

```diff
@@ -0,0 +1,97 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceValueUO_QP.h"
+#include "MooseMesh.h"
+registerMooseObject("MooseTestApp", InterfaceValueUO_QP);
+
+template <>
+InputParameters
+validParams<InterfaceValueUO_QP>()
+{
+  InputParameters params = validParams<InterfaceValueUserObject>();
+  params.addRequiredCoupledVar("var", "The variable name");
+  params.addCoupledVar("var_neighbor", "The variable name");
+  params.addClassDescription("Test Interfae User Object computing and storing average values at "
+                             "each QP across an interface");
+  return params;
+}
+
+InterfaceValueUO_QP::InterfaceValueUO_QP(const InputParameters & parameters)
+  : InterfaceValueUserObject(parameters),
+    _u(coupledValue("var")),
+    _u_neighbor(parameters.isParamSetByUser("var_neighbor") ? coupledNeighborValue("var_neighbor")
+                                                            : coupledNeighborValue("var"))
+
+{
+}
+
+InterfaceValueUO_QP::~InterfaceValueUO_QP() {}
+
+void
+InterfaceValueUO_QP::initialize()
+{
+  // define the boundary map and retrieve element side and boundary_ID
+  std::vector<std::tuple<dof_id_type, unsigned short int, boundary_id_type>> elem_side_bid =
+      _mesh.buildSideList();
+
+  // retrieve on which boundary this UO operates
+  std::set<BoundaryID> boundaryList = boundaryIDs();
+
+  // clear map values
+  _map_values.clear();
+
+  // initialize the map_values looping over all the element and sides
+  for (unsigned int i = 0; i < elem_side_bid.size(); i++)
```

Comment:
We are working hard to try and eliminate the use of `unsigned int` when looping over STL containers because the type returned by the `size()` method is `std::size_t`. We have a helper macro that we've recently added that makes it so you never have to think about the type used for your index. Make sure that this is rebased onto the most recent `next` branch pointer, and then you can use this syntax:
```c++
for (MooseIndex(elem_side_bid) i = 0; i < elem_side_bid.size(); i++)
```
Using this design pattern there will never be a mismatch between the type of `i` and the type of `elem_side_bid.size()`. If you want to know about the macro you can search for `MooseIndex` in `MooseTypes.h`